File: combat/src/classicthunder/combat/layout/DeckLayout.java
package classicthunder.combat.layout;

import classicthunder.combat.card.Position;
import com.badlogic.gdx.math.Vector2;
import com.badlogic.gdx.utils.viewport.Viewport;

public class DeckLayout {

    final private int worldWidth = 1600;
    final private int centerWidth = worldWidth / 2;

    final private int worldHeight = 900;
    final private int centerHeight = worldHeight / 2;

    final private int edgeBuffer = 200;

    final private Position drawPosition;
    final private Position discardPosition;

    final private Viewport viewport;

    public DeckLayout(Viewport viewport, Vector2 drawLocation, Vector2 discardLocation) {

        this.viewport = viewport;

        this.drawPosition = new Position(drawLocation, new Vector2(100, 150), 0f);
        this.discardPosition = new Position(discardLocation, new Vector2(100, 150), 0f);
    }

    public Position getDrawPosition() {
        return drawPosition;
    }

    public Position getDiscardPosition() {
        return discardPosition;
    }

    public Viewport getViewport() {
        return viewport;
    }

    public int getCenterWidth() {
        return centerWidth;
    }

    public int getCenterHeight() {
        return centerHeight;
    }

    public int getHandWidth() {
        return worldWidth - (2 * edgeBuffer);
    }
}

File: combat/src/classicthunder/combat/layout/DeckLayout.kt
package classicthunder.combat.layout

import classicthunder.combat.card.Position
import com.badlogic.gdx.math.Vector2
import com.badlogic.gdx.utils.viewport.Viewport

class DeckLayout(var viewport: Viewport, drawLocation: Vector2, discardLocation: Vector2) {

    private val worldWidth = 1600f
    val centerWidth = worldWidth / 2f

    private val worldHeight = 900f
    val centerHeight = worldHeight / 2f

    private val edgeBuffer = 200f
    val handWidth = worldWidth - 2 * edgeBuffer

    internal val drawPosition: Position = Position(drawLocation, Vector2(100f, 150f), 0f)
    internal val discardPosition: Position = Position(discardLocation, Vector2(100f, 150f), 0f)
}